src/Microsoft.Extensions.SecretManager.Tools/Program.cs

```diff
@@ -142,52 +140,48 @@ internal int RunInternal(params string[] args)
 
         private string ResolveUserSecretsId(CommandLineOptions options)
         {
+            if (!string.IsNullOrEmpty(options.Id))
+            {
+                return options.Id;
+            }
+
             var projectPath = options.Project ?? _workingDirectory;
 
             if (!Path.IsPathRooted(projectPath))
             {
                 projectPath = Path.Combine(_workingDirectory, projectPath);
             }
 
-            if (!projectPath.EndsWith("project.json", StringComparison.OrdinalIgnoreCase))
+            var directoryInfo = new DirectoryInfo(projectPath);
+            FileInfo fileInfo;
+            if (directoryInfo.Exists)
             {
-                projectPath = Path.Combine(projectPath, "project.json");
+                var projects = directoryInfo.EnumerateFileSystemInfos("*.csproj").ToList();
```

Comment:
We should normalize all this logic across our "complex" tools. Also, your repo where you normalize project.json and MSBuild project reading seem like the correct approach to move forward. Aka, our tools need to continue supporting both project.json and MSBuild for a little while, might as well make the process painless for all 😄